source/common/router/router.cc

```diff
@@ -704,8 +704,31 @@ Http::FilterHeadersStatus Filter::decodeHeaders(Http::RequestHeaderMap& headers,
   // Hang onto the modify_headers function for later use in handling upstream responses.
   modify_headers_ = modify_headers;
 
-  UpstreamRequestPtr upstream_request =
-      std::make_unique<UpstreamRequest>(*this, std::move(generic_conn_pool));
+  bool as_early_data{false};
```

Comment:
nit: I'm not quite sure what "as_early_data" means.

Reply:
rename to `can_use_early_data`